pkg/volume/csi/csi_mounter.go

```diff
@@ -279,7 +279,9 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
 	}
 
 	if c.supportsFSGroup(fsType, mounterArgs.FsGroup, c.fsGroupPolicy) {
-		err := volume.SetVolumeOwnership(c, mounterArgs.FsGroup, mounterArgs.FSGroupChangePolicy, util.FSGroupCompleteHook(c.plugin.GetPluginName()))
+		// fullPluginName helps to distinguish different driver from csi plugin
+		fullPluginName := fmt.Sprintf("%s:%s", c.plugin.GetPluginName(), c.driverName)
+		err := volume.SetVolumeOwnership(c, mounterArgs.FsGroup, mounterArgs.FSGroupChangePolicy, util.FSGroupCompleteHook(fullPluginName))
 		if err != nil {
```

Comment:
Can we fix `GetFullQualifiedPluginNameForVolume` instead to consider `CSIVolumeSource` correctly?

Reply:
updated.